### PR TITLE
Trivia: Use an Object for questions instead of a sorted array

### PR DESCRIPTION
**Pull request by AnnikaCodes**
This will make future features (like custom categories) much simpler, and eliminate odd behaviors, like the `ALL_CATEGORIES` object requiring alphabetically-sorted keys.

### CODE REVIEW

File: server/chat-plugins/trivia.ts [hunk 1]
```diff
@@ -1793,19 +1770,22 @@ const triviaCommands: ChatCommands = {
...
-			`<tr><td colspan="6"><strong>${Chat.count(submissionsLen, "</strong> questions")} awaiting review:</td></tr>` +
-			`<tr><th>#</th><th>${this.tr`Category`}</th><th>${this.tr`Question`}</th><th>${this.tr`Answer(s)`}</th><th>${this.tr`Submitted By`}</th></tr>`;
 
+		let innerBuffer = '';
 		let i = 0;
-		while (i < submissionsLen) {
-			const entry = submissions[i];
-			buffer += `<tr><td><strong>${(++i)}</strong></td><td>${entry.category}</td><td>${entry.question}</td><td>${entry.answers.join(", ")}</td><td>${entry.user}</td></tr>`;
+		for (const category in submissions) {
+			for (const entry of submissions[category]) {
```
Comment (mia-pi-git) [inline server/chat-plugins/trivia.ts:1777]:
```suggestion
			for (const [i, entry] of submissions[category].entries()) {
```

File: server/chat-plugins/trivia.ts [hunk 2]
```diff
@@ -1793,19 +1770,22 @@ const triviaCommands: ChatCommands = {
...
+				i++;
+				innerBuffer += `<tr><td><strong>${submissions[category].indexOf(entry)}</strong></td><td>${entry.category}</td><td>${entry.question}</td><td>${entry.answers.join(", ")}</td><td>${entry.user}</td></tr>`;
```
Comment (mia-pi-git) [inline server/chat-plugins/trivia.ts:1779]: Use the 'i' from above suggestion here instead of indexOf.

File: server/chat-plugins/trivia.ts (hunk 1 above)
Comment (AnnikaCodes) [reply server/chat-plugins/trivia.ts:1777]: We need `i` outside of the loop, on line 1785. (It's the total number of questions in all categories that await review.)

File: server/chat-plugins/trivia.ts (hunk 2 above)
Comment (AnnikaCodes) [reply server/chat-plugins/trivia.ts:1779]: Oh, I see what you mean, separating the index from the total.

File: server/chat-plugins/trivia.ts (hunk 1 above)
Comment (mia-pi-git) [reply server/chat-plugins/trivia.ts]: Call it index or something, then?

Comment (mia-pi-git) [reply server/chat-plugins/trivia.ts]: Or tbh maybe just rename the other variable, since "i" seems unclear for that.

Comment (AnnikaCodes) [reply server/chat-plugins/trivia.ts]: Yes, I named it `total` instead.

**Comment by Zarel** [comment]
I will trust that this is a good change, and also back up Trivia data just in case.

**Pull request merged**